### PR TITLE
Feature/post endpoint

### LINKED ISSUE #47: Implement PUT path in server

### LINKED ISSUE #49: Implement POST path on server

### LINKED ISSUE #50: Implement POST in client side to add journal

### PR DESCRIPTION
**Pull request by espressoGoddess**
### Files Changed:
- Activities.js
- App.js
- api-calls.js
- server.js

### What Has Changed: 
- This builds out the endpoints for POST and PUT on server side code.
- This implements POST request in client side!!! (method added in `api-calls` and implemented in `App`).
- This gets rid of `activities` from state (since we never change it, we don't really need to keep track of it) and instead passes `activityData` to `Activities` component.
- This also gets rid of `activity` from `App`'s state since we call this function upon clicking on `Let's do it` button when selecting an activity.

### Notes: 
- Once we have all the api-calls methods in, we could refactor to be one method (Joe opened an issue).

### Closes: 
Closes #47, closes #49, closes #50

### CODE REVIEW

File: client/src/Components/App/App.js [hunk 1]
```diff
@@ -17,24 +17,30 @@ const App = () => {
...
+  const addEntry = (activity) => {
+    const date = new Date();
+    const newEntry = {
+      user_id: user.id,
+      date: date.toISOString().split('T')[0],
+      feeling: feeling,
+      activity: activity,
+      journal_entry: journal
+    }
+    addNewEntry(newEntry).then(getAllEntries).then(data => setUserLogs(data));
```
Comment (chrissycooper) [inline client/src/Components/App/App.js:36]: this is SO CLEAN I'm obsessed. Let's go, perfectly up-to-date data model!!

File: server/server.js [hunk 2]
```diff
@@ -23,4 +23,25 @@ app.get('/api/v1/entries', async (req, res) => {
...
+app.put('/api/v1/entries/:id', async (req, res) => {
+  try {
+    const { id } = req.params;
+    const { journal_entry } = req.body;
+    await knex("entries").where({ entry_id: id }).update({ journal_entry: journal_entry });
+      res.status(204).json(`Entry ${id} was updated successfully`);
+  } catch (error) {
+    console.error(error.message);
+  }
+})
```
Comment (chrissycooper) [inline server/server.js:47]: Proud of us

**Review by chrissycooper — APPROVED** [review]

File: client/src/Components/App/App.js (hunk 1 above)
Comment (jfogiato) [reply client/src/Components/App/App.js:36]: +1 - Super duper fantastic special work here from everyone!!!

File: client/src/Components/App/App.js [hunk 3]
```diff
@@ -17,24 +17,30 @@ const App = () => {
...
+  }
+
   const filterActivities = () => {
     const feelingCategory = feelingsData
       .find(({ associatedFeelings }) => associatedFeelings.includes(feeling))
       ?.feeling.toLowerCase();
-    const filtered = activities.filter(act => act.feelings.includes(feelingCategory));
```
Comment (jfogiato) [inline client/src/Components/App/App.js:37]: YEET

File: client/src/Components/App/App.js [hunk 4]
```diff
@@ -17,24 +17,30 @@ const App = () => {
   const [user, setUser] = useState({id: 1, name: 'KB'});
   const [userLogs, setUserLogs] = useState([]);
   const [feeling, setFeeling] = useState('');
-  const [activity, setActivity] = useState('');
```
Comment (jfogiato) [inline client/src/Components/App/App.js:20]: YEET

File: client/src/Components/App/App.js [hunk 5]
```diff
@@ -17,24 +17,30 @@ const App = () => {
...
   const [journal, setJournal] = useState('');
-  const [activities, setActivities] = useState(activityData);
```
Comment (jfogiato) [inline client/src/Components/App/App.js:22]: YEET

File: client/src/Components/App/App.js [hunk 6]
```diff
@@ -17,24 +17,30 @@ const App = () => {
...
   const [filteredActivities, setFilteredActivities] = useState([]);
 
-  const updateActivity = (activity) => {
-    setActivity(activity);
-  }
-
```
Comment (jfogiato) [inline client/src/Components/App/App.js:28]: YEET

File: client/src/Components/App/App.js [hunk 7]
```diff
@@ -47,8 +53,6 @@ const App = () => {
     .then(data => setUserLogs(data));
   }, []);
 
-  
-
```
Comment (jfogiato) [inline client/src/Components/App/App.js:51]: YEET

File: server/server.js (hunk 2 above)
Comment (jfogiato) [reply server/server.js:47]: So proud.

**Review by jfogiato — APPROVED** [review]
Dora The Explorer's "We Did It!" song has been stuck in my head throughout this entire PR review.

So proud. Such good work and I'm so glad that we're all getting there together. Team JACK 4 life!!

File: client/src/Components/App/App.js [hunk 8]
```diff
@@ -17,24 +17,30 @@ const App = () => {
...
-  
   const saveNewEntry = (newEntry) => {
     setJournal(newEntry);
   }
 
+  const addEntry = (activity) => {
+    const date = new Date();
+    const newEntry = {
+      user_id: user.id,
+      date: date.toISOString().split('T')[0],
```
[the remark this answers is not in the document]
Comment (KatherineBlaine) [reply client/src/Components/App/App.js:31]: Did we ever find out what 'T' means?

File: client/src/Components/Activities/Activities.js [hunk 9]
```diff
@@ -21,7 +21,7 @@ const Activities = ({ activities, updateActivity, setFeeling }) => {
   }
 
   const selectActivity = () => {
-    updateActivity(activities[currentActivityIndex].activity);
+    addEntry(activities[currentActivityIndex].activity);
   }
 
   return (
```
Comment (KatherineBlaine) [inline client/src/Components/Activities/Activities.js:27]: So much cleaner! Thanks for working on this.

File: client/src/Components/App/App.js (hunk 1 above)
Comment (KatherineBlaine) [reply client/src/Components/App/App.js:36]: SO clean!

File: server/server.js (hunk 2 above)
Comment (KatherineBlaine) [reply server/server.js:47]: YAY!! Good work team!

**Review by KatherineBlaine — APPROVED** [review]

**Pull request merged**